### PR TITLE
Rename Template to DataModel and clean up

### PR DESCRIPTION
**Pull request by mfucci**

### CODE REVIEW

**Comment by Apollon77** [comment]
Ok I will wait with my further cluster definitzion PRs before this is ready and merged because I need again to adjust a lot and easier locally :-)

**Review by turon — COMMENTED** [review]
Some initial thoughts on the renaming.  Definitely agree that renaming will improve readability.

File: src/codec/DataModels.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    tlvType?: TlvType,
+    postProcess?: (value: any) => T,
+    preProcess?: (value: T) => any,
+    validator?: (value: T, name: string, model: DataModel<T>) => void,
+};
+
+type FixedOrVariableLengthConstraints = { length: number } | { minLength: number, maxLength: number };
+
+// Define array data model
+export type ArrayM<T> = Merge<{ tlvType: TlvType.Array, itemTemplate: DataModel<T> }, FixedOrVariableLengthConstraints>;
```
[the remark this answers is not in the document]
Comment (Apollon77) [reply src/codec/DataModels.ts:21]: would agree naming wise, alsoto keep in sync with UIntXT and others

**Comment by Apollon77** [comment]
@mfucci Should I wait with my cluster PRs for this to be finished or what do you think?

**Comment by mfucci** [comment]
Abandoning this PR in favor of Matter.js schema: templates in node-matter will be refactored to use the TLV codex fro matter.js

**Pull request closed without merging**